freeipa/src/main/java/com/sequenceiq/freeipa/converter/stack/StackToCreateFreeIpaRequestConverter.java

```
@@ -0,0 +1,409 @@
+            }
+        }
+        LOGGER.debug("Created tags request {} from tags {}", userDefined, tags);
+        return userDefined;
+    }
+
+    private String getLoggingLocation(String location) {
+        return getLocation(location, CLUSTER_LOG_PREFIX);
+    }
+
+    String getBackupLocation(Stack stack, String location) {
+        String originalLocation = getLocation(location, CLUSTER_BACKUP_PREFIX);
+
+        // During provisioning, the URL for Azure is converted from abfs/abfss to https, this conversion needs to be reversed.
+        // Example: convert https://storage1.dfs.core.windows.net/logs-fs into abfs://logs-fs@storage1.dfs.core.windows.net
+        try {
+            if (CloudPlatform.AZURE.equalsIgnoreCase(stack.getCloudPlatform()) && originalLocation != null) {
+                URI uri = new URI(originalLocation);
+                if (AZURE_BLOB_STORAGE_SCHEMA.equals(uri.getScheme())) {
+                    String uriPath = uri.getPath();
+                    int firstSeparator = uriPath.indexOf(PATH_DELIMETER);
+                    if (firstSeparator != -1) {
+                        int secondSeparator = uriPath.indexOf(PATH_DELIMETER, firstSeparator + 1);
+                        String bucketName;
+                        String bucketPath = "";
+                        if (secondSeparator == -1) {
+                            bucketName = uriPath.substring(firstSeparator + 1);
+                        } else {
+                            bucketName = uriPath.substring(firstSeparator + 1, secondSeparator);
+                            bucketPath = uriPath.substring(secondSeparator);
+                        }
+                        originalLocation = String.format("%s%s@%s%s", ORIGINAL_AZURE_BLOB_STORAGE_SCHEMA, bucketName, uri.getHost(), bucketPath);
+                    }
+                }
+            }
+        } catch (URISyntaxException e) {
+            LOGGER.error("URI syntax error");
```

are you sure we should ignore this error and move on with the rebuild?

---

I changed it to throw an exception.